lib/GADS.pm

```diff
@@ -4796,7 +4807,13 @@ sub _process_edit
         {
             # The "source" parameter is user input, make sure still valid
             my $source_curval = $layout->column(param('source'), permission => 'read');
-            try { $record->write(dry_run => 1, parent_curval => $source_curval) };
+            my %options = (dry_run => 1, parent_curval => $source_curval);
+            # If this is being submitted as part of an autosave recovery, allow
+            # mandatory values to be empty. This is because a field may have
+            # since been made mandatory, and we don't want this to cause the
+            # autosave recovery to fail
+            $options{force_mandatory} = 1 if param 'autosave';
```

Comment:
My concern about this is that there is then a user way to always override mandatory (if they know what they are doing) - then can just add this parameter to any form submission. We probably need a way of tightening this, or preferably there would be a way of notifying the user that a saved value is now mandatory?

Reply:
If I'm honest, I'm having some trouble with this one. I don't quite know whether it would be best to check the fields are mandatory (and blank) in the JS and provide feedback in the auto-recover modal or to have a better check (maybe using the `user_key` rather than just checking the param is 1) backend to ensure this is only called in the correct circumstances. Suggestions welcomed!!